www/js/lib/uiUtil.js

```diff
@@ -438,14 +441,21 @@ define(rqDef, function() {
         // is not dark (but we want it applied when the content is dark or inverted)
         footer.classList.add(contentTheme || '_light');
         // Embed a reference to applied theme, so we can remove it generically in the future
-        htmlEl.dataset.theme = theme;
+        htmlEl.dataset.theme = appTheme + contentTheme;
         // Hide any previously displayed help
-        var oldHelp = document.getElementById(oldTheme + '-help');
+        var oldHelp = document.getElementById(oldContentTheme.replace(/_/, '') + '-help');
         if (oldHelp) oldHelp.style.display = 'none';
         // Show any specific help for selected contentTheme
-        var help = document.getElementById(theme + '-help');
+        var help = document.getElementById(contentTheme.replace(/_/, '') + '-help');
         if (help) help.style.display = 'block';
-        
+        // Remove the contentTheme for auto themes whenever system is in light mode
+        if (/^auto/.test(theme) && appTheme === 'light') contentTheme = null;
+        // Hide any previously displayed description for auto themes
+        var oldDescription = document.getElementById('kiwix-auto-description');
+        if(oldDescription) oldDescription.style.display = 'none';
```

Comment:
Please add a space between `if` and `(oldDescription)` and we'll start the process of merging this PR.
There is a conflict with master that I will resolve